server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -295,15 +295,15 @@ private DocumentMapper newDocumentMapper(Mapping mapping, MergeReason reason) {
         return newMapper;
     }
 
-    private Mapping parseMappings(String mappingType, CompressedXContent mappingSource) {
+    public Mapping parseMapping(String mappingType, CompressedXContent mappingSource) {
         try {
             return mappingParser.parse(mappingType, mappingSource);
         } catch (Exception e) {
             throw new MapperParsingException("Failed to parse mapping: {}", e, e.getMessage());
         }
     }
 
-    private static Mapping mergeMappings(DocumentMapper currentMapper, Mapping incomingMapping, MergeReason reason) {
+    public static Mapping mergeMappings(DocumentMapper currentMapper, Mapping incomingMapping, MergeReason reason) {
```

Comment:
Does it make more sense to take two Mapping objects, rather than one Mapping and one DocumentMapper?  And if so maybe the method should like directly on Mapping as well?

Reply:
On taking Mapping, I was reluctant to have to handle the null document mapper outside of it, and then provide null in that case. It feels like we would end up handling null values in two places.

I also agree that this is not the right place for this method also. I think that this method is here rather temporarily, and will be moved once we introduce the mapping merger and reorganize the different merge methods on MapperService. That said I was reluctant to spend more time think about how to shape this method :)

All this said: is it reasonable to get back to this as a followup? I am afraid that however we shape it and move it, we will move it again very soon.

Reply:
OK, that all sounds reasonable.